src/utils/ComponentsTypes.ts

```diff
@@ -32,4 +36,11 @@ export class ComponentsTypes {
   public static getAllFacetsInstance(root: HTMLElement | Dom) {
     return ComponentsTypes.getAllFacetsElements(root).map(element => Component.get(element) as Component);
   }
+
+  public static getAllFacetsFromSearchInterface(searchInterface: IComponentsTypesSearchInterface) {
+    return ComponentsTypes.allFacetsType.reduce(
+      (facets: Component[], facetType: string) => facets.concat(searchInterface.getComponents(facetType)),
+      []
+    );
```

Comment:
Would that be equivalent to the following?
```ts
return _.flatten(
  ComponentsTypes.allFacetsType.map(facetType => searchInterface.getComponents(facetType))
);
```
If so, I think a map and flatten would be easier to read.

Reply:
I don't think the reduce() that complicated (or more complicated than the flatten), it's a bit more optimized and I don't need to type the return value. 
I could also just do a forEach() and push to an array.